BackEnd/scripts/updateStatsForLeague.ts

```diff
@@ -0,0 +1,58 @@
+import dotenv from 'dotenv';
+dotenv.config({ path: '../.env.development' });
+import { SaveDataByMatchId } from '../src/business/games';
+import { ToMatchId } from '../../Common/utils';
+import { GetOrCreatePlayerOverviewByName } from '../src/business/player';
+import { GetDbActiveSeasonWithSheets } from '../src/db/season';
+import { GetGoogleSheet } from '../src/external-api/sheets';
+import { buildRisenTeams, RisenSheetParser, RisenTeam } from '../src/business/teams';
+import { PremadeParser } from '../src/business/sheets/premadeParser';
+import { GetRiotGamesByPlayerPuuid } from '../src/external-api/game';
+
+
+/**
+ * WARNING: This function is only made for the premade leagues
+ * @param seasonId
+ */
+async function updatePlayerStatsForGame(seasonId: number) {
+  await buildRisenTeams(seasonId);
+
+  let seasonsWithSheets = await GetDbActiveSeasonWithSheets(seasonId);
+
+  let risenSheetTeams: RisenTeam[] = [];
+  for (let seasonsWithSheet of seasonsWithSheets) {
+    console.log(`found season ${seasonsWithSheet.seasonName}`);
+    let sheet = await GetGoogleSheet(seasonsWithSheet.googleSheetId, 'Teams and Standings');
+    let parser: PremadeParser =  new PremadeParser();
+    risenSheetTeams = parser.getRisenSheetTeams(sheet);
+  }
+
+  let players = [];
+  for (let risenSheetTeam of risenSheetTeams) {
+    players.push(await GetOrCreatePlayerOverviewByName(risenSheetTeam.top));
+    players.push(await GetOrCreatePlayerOverviewByName(risenSheetTeam.jungle));
+    players.push(await GetOrCreatePlayerOverviewByName(risenSheetTeam.mid));
+    players.push(await GetOrCreatePlayerOverviewByName(risenSheetTeam.adc));
+    players.push(await GetOrCreatePlayerOverviewByName(risenSheetTeam.support));
+  }
+
+  console.log(`Found ${players.length} players`);
+
+  let gameIds = new Set<string>;
+  for (let player of players) {
+    let games = await GetRiotGamesByPlayerPuuid(player.puuid, 6, true); // Use 10 incase there was rescedules
```

Comment:
Was 10 taking too long?

Reply:
I assumed that i was gonna run this script every week. The assumption is that i would only miss at max 2 series (2x3) if i did it per week.